railties/test/generators/actions_test.rb

```diff
@@ -51,7 +51,7 @@ def test_add_source_with_block_adds_source_to_gemfile_with_gem
     action :add_source, "http://gems.github.com" do
       gem "rspec-rails"
     end
-    assert_file "Gemfile", /source 'http:\/\/gems\.github\.com' do\n  gem 'rspec-rails'\nend/
+    assert_file "Gemfile", /source 'http:\/\/gems\.github\.com' do\n  gem 'rspec-rails'\nend\n\n/
```

Comment:
I think this should end with `\nend\n\z` (one newline not two, and then `\z` to prove there isn't another)

Reply:
Yeah you are right.

I fixed this with this commit.https://github.com/rails/rails/pull/35285/commits/d872920a434d8c86480ce913a3c9481d41f7dfcf ( I will squash all the commit into one before this gets merged)